packages/postgres-database/src/simcore_postgres_database/models/projects_tags.py

```diff
@@ -0,0 +1,27 @@
+import sqlalchemy as sa
+
+from .base import metadata
+from .projects import projects
+from .tags import tags
+
+projects_tags = sa.Table(
+    #
+    # Tags associated to a project (many-to-many relation)
+    #
+    "projects_tags",
+    metadata,
+    sa.Column(
+        "project_id",
+        sa.BigInteger,
+        sa.ForeignKey(projects.c.id, onupdate="CASCADE", ondelete="CASCADE"),
+        nullable=False,
+        doc="NOTE that project.c.id != project.c.uuid",
+    ),
+    sa.Column(
+        "tag_id",
+        sa.BigInteger,
+        sa.ForeignKey(tags.c.id, onupdate="CASCADE", ondelete="CASCADE"),
+        nullable=False,
+    ),
+    sa.UniqueConstraint("project_id", "tag_id"),
```

Comment:
ah is it so that you cannot have a primary key here because these are both foreigh keys? otherwise what is the advantage of unique constraint over primary key?

Reply:
this table is mapping two primary ids. It ensures that the combination of the specified columns must be unique across all rows in the table, but it does not necessarily identify rows uniquely (as a primary key does).

Reply:
I understand what unique means. I did some research in the train, and the only difference I found was that a primary key also does not allow for NULL, where unique constraint allows it. Otherwise they are equivalent.

Reply:
and in some cases you do not have the primary key defined at all.